runelite-client/src/main/java/net/runelite/client/plugins/customcursor/CustomCursorPlugin.java

```diff
@@ -67,13 +73,42 @@ protected void shutDown()
 	@Subscribe
 	public void onConfigChanged(ConfigChanged event)
 	{
-		if (event.getGroup().equals("customcursor") && event.getKey().equals("cursorStyle"))
+		if (event.getGroup().equals("customcursor"))
 		{
-			updateCursor();
+			if (event.getKey().equals("cursorStyle"))
+			{
+				useSelectedCursor();
+			}
+			else if (event.getKey().equals("customImage"))
+			{
+				updateCursor();
+			}
 		}
 	}
 
 	private void updateCursor()
+	{
+		// Custom image file takes precedent
```

Comment:
How's this UX? If you use the dropdown for an out of the box cursor, it sets it to that. However, on start up, if you have a custom image defined, it uses that.

Reply:
I felt like it would be more clear to have a distinct drop-down menu option for a "Custom" entry, which would use the file from the file select. That way, you would not need to reset the plugin's settings to stop using the custom file.

Reply:
But that would be weird if you choose a file and don't see it take effect right away, no?

Reply:
To stop using the custom file, you can change the dropdown part instead. Although then if you want to go back to using the custom file, you have to choose a new file or reset the config which is not ideal.

Reply:
You could set their selected dropdown config option to the custom cursor option when a file is selected